packages/@okta/vuepress-site/docs/guides/authenticators-custom-authenticator/main/index.md

```diff
@@ -0,0 +1,121 @@
+
+### Create an OIDC web authentication client
+
+The simplest way to integrate authentication in your app is with OIDC through a web browser, using the Authorization Code Flow grant. You need an access token to start the enrollment flow for the Devices SDK. For future logins, consider using refresh tokens. <StackSnippet snippet="samplecode" inline />
+
+### Grant the required scopes
+
+When you are ready to grant the required scopes, follow these steps:
+
+1. Sign in to your Okta organization with your administrator account and go to **Applications** > **Applications**.
+2. Open your OpenID Connect client app.
+3. On the **Okta API Scopes** tab, click **Grant** for the following scopes:
+   * For access to both GET and POST/DELETE endpoints:
+      * `okta.authenticators.manage.self`
+   * For access to GET endpoints only:
+      * `okta.authenticators.read`
+      * `okta.users.read.self`
+
+Alternatively, you can grant scopes using the [Grant consent to scope for application](/docs/reference/api/apps/#application-oauth-2-0-scope-consent-grant-operations) operation of the Apps API.
+
+### Set up notification services
+
+<StackSnippet snippet="notifserviceadminconsole" />
+
+### Add a custom authenticator
+
+<StackSnippet snippet="customauthenticatoradminconsole" />
+
+### Set up a global session policy and authentication policy
+
+You need to set up a Global Session Policy and an authentication policy to integrate with the Devices SDK. See [Configure a Global Session Policy and authentication policy](/docs/guides/configure-signon-policy/main/).
+
+## Install and configure the Okta Devices SDK
+
+<StackSnippet snippet="sdksteps" />
+
+## Limitations
+
+### Custom authenticator error in Admin Console
+
+#### Description
+
+Use cases that meet the following conditions receive an error message in the Admin Console UI:
+
+* Call the `api/v1/apps` endpoint to create an OIDC app with a custom `client_id`.
+* Use the Admin Console to create a custom authenticator.
+
+#### Workaround
+
+Create both the OIDC client app (with a custom `client_id`) and the custom authenticator in the Admin Console.
+
+Alternatively, you can call the `api/v1/apps` endpoint to create the OIDC app and `custom client_id`, and call the `api/v1/authenticators` endpoint to create a custom authenticator.
+
+## Troubleshoot
+
+If your push notifications aren't delivering:
+
+1. Follow the steps to [view push notification events]((https://help.okta.com/okta_help.htm?type=oie&id=ext-all-notification-services).
+2. To further narrow your search, enter the following: `eventType eq "device.push.provider.update" and displayMessage eq "Push Provider Configuration verification failed"`. See [Event types](/docs/reference/api/event-types/) for more details.
+3. In the **Reason** section, locate the error message from your push provider. Consult the push provider documentation, if necessary.
+4. You might need to upload a new JSON file if it has small errors, like typos.
```

Comment:
In Step 4, we should direct the admin to to go the Notification Services page in the admin console, and edit the respective notification service. We can also link documentation to the `Edit a notification service` section (same doc link as `view push notification events`)